pkg/feature/api/feature.go

```diff
@@ -1816,9 +1851,68 @@ func (s *FeatureService) EvaluateFeatures(
 		}
 		return nil, dt.Err()
 	}
+	// If the feature ID is set, it will return a single evaluation
+	if req.FeatureId != "" {
+		eval, err := s.findEvaluation(userEvaluations.Evaluations, req.FeatureId)
+		if err != nil {
+			s.logger.Error(
+				"Failed to find evaluation",
+				log.FieldsFromImcomingContext(ctx).AddFields(
+					zap.Error(err),
+					zap.String("environmentNamespace", req.EnvironmentNamespace),
+				)...,
+			)
+			dt, err := statusInternal.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.InternalServerError),
+			})
+			if err != nil {
+				return nil, statusInternal.Err()
+			}
+			return nil, dt.Err()
+		}
+		return &featureproto.EvaluateFeaturesResponse{
+			UserEvaluations: &featureproto.UserEvaluations{
+				Id:          userEvaluations.Id,
+				Evaluations: []*featureproto.Evaluation{eval},
+				CreatedAt:   userEvaluations.CreatedAt,
+			}}, nil
+	}
 	return &featureproto.EvaluateFeaturesResponse{UserEvaluations: userEvaluations}, nil
 }
 
+func (s *FeatureService) getTargetFeatures(
```

Comment:
Can you add the test case like https://github.com/bucketeer-io/bucketeer/blob/main/pkg/gateway/api/api_grpc_test.go#L2156?

Although I don't add it in the api.go because we'll remove it. It's worth to add the test case here.